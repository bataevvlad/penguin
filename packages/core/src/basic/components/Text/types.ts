import {TextProps as DefaultTextProps, Text as DefaultText} from 'react-native';

export type ITextProps = DefaultTextProps;
export type IText = DefaultText;
