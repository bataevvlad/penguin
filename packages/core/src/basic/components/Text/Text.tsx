import {forwardRef} from 'react';
import {Text as DefaultText} from 'react-native';

import {getConfig} from '../../config';

import {IText, ITextProps} from './types';

// eslint-disable-next-line react/display-name
const Text = forwardRef<IText, ITextProps>((props, ref) => {
  return (
    <DefaultText
      ref={ref}
      allowFontScaling={getConfig().allowFontScaling}
      {...props}
    />
  );
});
export default Text;
