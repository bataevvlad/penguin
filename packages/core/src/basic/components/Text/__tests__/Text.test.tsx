import {render} from '@testing-library/react-native';

import Text from '../Text';

describe('@penguin/core/Text', () => {
  it('renders correctly', () => {
    const {toJSON} = render(<Text>Text</Text>);
    expect(toJSON()).toMatchSnapshot();
  });
});
