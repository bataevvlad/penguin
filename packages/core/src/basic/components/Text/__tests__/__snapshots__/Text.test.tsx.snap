// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@penguin/core/Text renders correctly 1`] = `
<Text
  allowFontScaling={true}
>
  Text
</Text>
`;
